sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -685,13 +685,7 @@ def test_setting_request_on_sub_estimator_removes_error(metaestimator):
             )
             if "fit" not in method_name:
                 # fit before calling method
-                set_requests(
```

Comment:
Can you explain here why we don't need to call fit with the metadata. I'm a bit lost.

Reply:
This is something this PR is fixing. We don't have to pass metadata during `fit`. when what we're testing is `predict` for instance. Now with this PR things are stored separately instead of being overriden on top of one another, so we can skip passing metadata to `fit` when it's not necessary.

Reply:
OK it makes sense then.